### ISSUE TITLE
java.util.concurrent.RejectedExecutionException when queue looks empty

### ISSUE THREAD

**Issue by hvesalai** [slick/slick#1827]
I am seeing this:

```
java.util.concurrent.RejectedExecutionException: Task slick.basic.BasicBackend$DatabaseDef$$anon$2@3cb503be rejected from slick.util.AsyncExecutor$$anon$2$$anon$1@48bf4020[Running, pool size = 10, active threads = 5, queued tasks = 0, completed tasks = 753206]
```

Looks like a task is being rejected even when the queue size is 0, i.e. there is nothing in the queue

**Comment by MartynasSapoka** [comment]
Having the same issue. Looks like slick deadlocks and does not return the threads to the pool

**Comment by hvesalai** [comment]
The problem here is that the queue is not actually empty. It is paused and the code is showing it as empty when in that state.

**Comment by hvesalai** [comment]
Fixed by #1854
